ENGAGEHF/Dashboard/Vitals/VitalsManager.swift

```diff
@@ -0,0 +1,353 @@
+//
+// This source file is part of the ENGAGE-HF project based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import FirebaseAuth
+import FirebaseFirestore
+import Foundation
+import HealthKit
+import struct ModelsR4.DateTime
+import class ModelsR4.ObservationComponent
+import OSLog
+import Spezi
+import SpeziFirebaseConfiguration
+
+
+/// Vitals History Manager
+///
+/// Functionality:
+/// - Maintain local, up-to-date arrays of the patients health data
+/// - Convert FHIR quantities to HK Samples
+/// - Manage units of the stored quantities, defaulting to the user's phone settings
+@Observable
+public class VitalsManager: Module, EnvironmentAccessible {
+    enum VitalsError: Error {
+        case invalidConversion
+        case unknownUnit
+        case invalidObservationType
+        case missingField
+    }
+    
+    
+    @ObservationIgnored @Dependency private var configureFirebaseApp: ConfigureFirebaseApp
+    @ObservationIgnored @StandardActor private var standard: ENGAGEHFStandard
+    private let logger = Logger(subsystem: "ENGAGEHF", category: "VitalsManager")
+    
+    private var authStateDidChangeListenerHandle: AuthStateDidChangeListenerHandle?
+    private var snapshotListener: ListenerRegistration?
+    
+    // TODO: Localize the units based on either locale or user preferences in the phone settings
```

Comment:
I think it is fine if they are not localized here as long as we do that as part of the display logic.